### PR TITLE
[T3A3][W13-A1] Conan Kian

### PR DESCRIPTION
**Pull request by ckjr**
Ready for review

### CODE REVIEW

**Comment by okkhoy** [comment]
@ckjr 
Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

File: src/seedu/addressbook/data/person/Block.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package seedu.addressbook.data.person;
+
+/**
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: good! you have a header comment

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -10,23 +10,56 @@
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String MESSAGE_ADDRESS_INVALID = "Person address must be in the following format a/BLOCK, STREET, UNIT, POSTAL_CODE ";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: you need to update the user guide as well to reflect the changes,

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -10,23 +10,56 @@
...
+    public static final int STREET_NUMBER = 1;
+    public static final int UNIT_NUMBER = 2;
+    public static final int POSTALCODE_NUMBER = 3;
+    
+    public static final int REQUIRED_NUMBER_OF_ADDRESS_FIELDS = 4;
+    
     public final String value;
+    
+    
+    private final Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: think, why does making this `final` makes little sense?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -10,23 +10,56 @@
...
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+
         this.value = address;
+        
+        final String[] splitAddress = address.split(", ");
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: this split may not work if my address is "123,clementi ave3,#03-23,s123456"
- what is the better way to handle spaces here? 
- what would be better way to handle the split?

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -10,23 +10,56 @@
...
+       
+        if (splitAddress.length < REQUIRED_NUMBER_OF_ADDRESS_FIELDS) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_INVALID);
+        }
+        
+        this.block = new Block(splitAddress[BLOCK_NUMBER]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: good! no magic numbers :+1:

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -10,23 +10,56 @@
...
+        this.street = new Street(splitAddress[STREET_NUMBER]);
+        this.unit = new Unit(splitAddress[UNIT_NUMBER]);
+        this.postalCode = new PostalCode(splitAddress[POSTALCODE_NUMBER]);
+        
     }
+    
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: how will you check for object equality?

how will you rebuild the address  string from the components when necessary?

**Comment by okkhoy** [comment]
@ckjr 
Some comments added. Please ack & close the PR after reading comments.

This seems like an incomplete attempt to me.

File: src/seedu/addressbook/data/person/Address.java (hunk 3 above)
Comment (ckjr) [inline src/seedu/addressbook/data/person/Address.java]: Hmm, just realized that the address may be subject to edits after creation, so it should not be final.

File: src/seedu/addressbook/data/person/Address.java (hunk 4 above)
Comment (ckjr) [inline src/seedu/addressbook/data/person/Address.java]: Actually, I was under the assumption that the address input will be in the correct format.

To handle spaces, I would use
`address.split(", *");`

Also, I did not take into account that the address may be entered with one or more fields missing.

So I think I would first split it, and then count the number of fields in the string array. After that I would instantiate the correct field classes. (This is assuming that the fields are entered in the correct order still, just that some fields are missing.)

File: src/seedu/addressbook/data/person/Address.java (hunk 6 above)
Comment (ckjr) [inline src/seedu/addressbook/data/person/Address.java]: Add a compareTo function to all the address field classes. First the name of the field will be compared through the use of strings (I.e. "Block"), and then if the type matches, the content of the field will then be compared.

I will add the function that returns this
`block.getBlock() + ", " + street.getStreet() + ", " + unit.getUnit() + ", " + postalCode.getPostalCode()`

**Comment by ckjr** [comment]
Acknowledged.

**Pull request closed without merging**